### DISCUSSION TITLE
Segmentation Fault

### DISCUSSION THREAD

**Discussion by wthoutanymmries** [uNetworking/uWebSockets.js#385] [Q&A] [answered]
...

**Comment by hst-m** [comment]
The [Readme](https://github.com/uNetworking/uWebSockets.js#package-easily-installed) says to install with `npm install uNetworking/uWebSockets.js#v18.9.0` have you tried that? Why are you trying to compile it if you have no idea what you are doing? Don't follow that guys tutorial. There is Pub/Sub example [here](https://github.com/uNetworking/uWebSockets.js/blob/master/examples/PubSub.js) so what do you mean there is no example? install with the line above and replace the 1st line in the example with `const uWS = require('uWebSockets.js')`

**Comment by wthoutanymmries** [comment]
@hst-m  Yeah i tried like a dozen of different versions, tried to reinstall node several times and it just does not work
Found [this one](https://github.com/uNetworking/pubsub-example) tho
It actually has a client but the one you mentioned does not that what I meant
Feel free to correct me I'm here to learn

**Comment by hst-m** [comment]
> Yeah i tried like a dozen of different versions, tried to reinstall node several times and it just does not work

you have provided no info on how what does not work

> It actually has a client but the one you mentioned does not that what I meant
Feel free to correct me I'm here to learn

websocket client is not too hard just google it or ask if you need help

**Comment by wthoutanymmries** [comment]
I don't ask you to teach me
I sound like a noob because I am one
uWS.App() and uWS.SSLApp() were crashing with segmentation fault as soon as I made connection
To my eye there is no mistake in my code and clearly if the error was segmentation fault then something was trying to access something that was already deleted or something was going out of bounds
It was a mistake in native code, not in js runtime
I tested multiple node versions so most likely it's not node
I reported what I found so may be it could help in some way
I do not have any questions
Feel free to close/delete this issue

**Comment by hst-m** [comment]
I don't use Mac to check and this is not my repo but I am saying if you want to report something you should be using the easy Readme install and using one of the included examples to make sure you are doing it correctly, and report the specifics of your problem, not doing your custom compile and using someone else's old library for the example

**Comment by wthoutanymmries** [comment]
@hst-m yeah) solution turned out to be embarrassingly simple

**Comment by ghost** [comment]
What was the solution?

**Comment by shaggyrec** [comment]
@wthoutanymmries I have "segmentation fault 11". What is your solution?

**Comment by wthoutanymmries** [comment, marked as answer]
@shaggyrec basic type checking. Make sure you pass strings not js objects

**Comment by shaggyrec** [comment]
I made `.ws(...).listen(parseInt(PORT), () => {})` and it works. `PORT` is the variable from env. It is string. 
It wasn't obvious.

**Comment by ghost** [comment]
https://github.com/uNetworking/uWebSockets.js/issues/343

**Comment by hst-m** [comment]
Ah the OP was using Publish with an object instead of stringify it first, looks like the tutorial repo he tried to use was bad https://github.com/wthoutanymmries/uws_test/blob/master/src/server/server.js#L59

**Comment by ghost** [comment]
I have added stricter type checks here https://github.com/uNetworking/uWebSockets.js/commit/cee33dc003a18407374a86a8440f8e84c825882e

**Comment by hst-m** [comment]
That looks like silent fail, and extra processing for every publish call just to make sure called correctly, when can just follow docs correctly

**Comment by ghost** [comment]
They are not silent,

https://github.com/uNetworking/uWebSockets.js/blob/master/src/AppWrapper.h#L338

There are now three checks that will throw exceptions with accurate info on wrong arguments passed.
